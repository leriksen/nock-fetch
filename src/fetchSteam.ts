// if you ask
// tslint:disable-next-line
// http://steamcommunity.com/market/priceoverview/?appid=730&currency=3&market_hash_name=StatTrak%E2%84%A2 M4A1-S | Hyper Beast (Minimal Wear)
//
// you'll get something like
// {"success":true,"lowest_price":"261,35&#8364; ","volume":"11","median_price":"269,52&#8364; "}

import nodeFetch from 'node-fetch';

export interface SteamPriceData {
  lowest_price: string;
  median_price: string;
  success: boolean;
  volume: string;
}

const generateDataUrl = (id: string, name: string) =>
  'https://steamcommunity.com/market/priceoverview/' +
  `?appid=${id}&country=DE&currency=3&market_hash_name=${encodeURIComponent(name.toUpperCase())}`;

export const getData = async (id: string, name: string): Promise<SteamPriceData> => {
  const dataUrl = generateDataUrl(id, name);
  const result = await nodeFetch(dataUrl);
  return await result.json();
};
