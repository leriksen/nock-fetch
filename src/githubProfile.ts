import nodeFetch from 'node-fetch';

const generateDataUrl = (name: string): string => {
  return `https://api.github.com/users/${name}`;
};

export const getData = async (name: string): Promise<string> => {
  const dataUrl = generateDataUrl(name);
  const result = await nodeFetch(dataUrl);
  return await result.json();
};
