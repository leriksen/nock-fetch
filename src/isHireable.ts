// queries github to see if the specifed user has indicated they are available for hire

import { default as fetch } from 'node-fetch';

const generateDataUrl = (name: string): string => {
  return `https://api.github.com/users/${name}`;
};

export const isHireable = async (name: string): Promise<boolean> => {
  const dataUrl = generateDataUrl(name);
  const result = await fetch(dataUrl);
  const json = await result.json();
  return json['hireable'] == true;
};
