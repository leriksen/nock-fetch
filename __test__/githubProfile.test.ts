import "jest";

import { setupRecorder } from "nock-record";
import { getData } from "../src/githubProfile";

const record = setupRecorder();

describe("githubPrfile", () => {
    describe("getData", () => {

        it("gets data on user from Github", async () => {
            const {completeRecording, assertScopesFinished} = await record("github-data");
            const result = await getData('leriksen');
            completeRecording();
            assertScopesFinished();
            expect(result).toMatchSnapshot();
        });
    });
});
