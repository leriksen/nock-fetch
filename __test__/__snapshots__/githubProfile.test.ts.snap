// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`githubPrfile getData gets data on user from Github 1`] = `
Object {
  "avatar_url": "https://avatars2.githubusercontent.com/u/441543?v=4",
  "bio": null,
  "blog": "",
  "company": null,
  "created_at": "2010-10-16T04:00:16Z",
  "email": null,
  "events_url": "https://api.github.com/users/leriksen/events{/privacy}",
  "followers": 0,
  "followers_url": "https://api.github.com/users/leriksen/followers",
  "following": 1,
  "following_url": "https://api.github.com/users/leriksen/following{/other_user}",
  "gists_url": "https://api.github.com/users/leriksen/gists{/gist_id}",
  "gravatar_id": "",
  "hireable": null,
  "html_url": "https://github.com/leriksen",
  "id": 441543,
  "location": "Bendigo, Victoria, Australia",
  "login": "leriksen",
  "name": "Leif Eriksen",
  "node_id": "MDQ6VXNlcjQ0MTU0Mw==",
  "organizations_url": "https://api.github.com/users/leriksen/orgs",
  "public_gists": 11,
  "public_repos": 64,
  "received_events_url": "https://api.github.com/users/leriksen/received_events",
  "repos_url": "https://api.github.com/users/leriksen/repos",
  "site_admin": false,
  "starred_url": "https://api.github.com/users/leriksen/starred{/owner}{/repo}",
  "subscriptions_url": "https://api.github.com/users/leriksen/subscriptions",
  "type": "User",
  "updated_at": "2019-02-26T13:51:43Z",
  "url": "https://api.github.com/users/leriksen",
}
`;
